leader_test.go

```diff
@@ -368,10 +365,14 @@ func TestLeader_divideHealthChecks(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	// We can't really tell which agent will pick up node3 with 100% certainty
+	// and we've already established that rebalancing worked higher up
+	//
+	// agent1.verifyUpdates(t, []string{"node1", "node3"}, []string{})
+	// agent2.verifyUpdates(t, []string{"node2"}, []string{})
+
```

Comment:
This makes a lot of sense. From the test failure messages you included in the description, it seems like `TestLeader_divideCoordinates` and `TestLeader_rebalanceHealthWatches` have a similar `verifyUpdates` check. Did you experience anything similar with these tests? I would have thought that this test is similar to those two but it looks like removing the `t.Parallel()` is sufficient for those two tests.

Reply:
It is odd.

But they are not testing exactly the same, in some cases the `probes` are being checked, in other places it is the `nodes`. I did not encounter additional problems with the other tests, but I may just have been lucky.